src/logger/src/metrics.rs

```diff
@@ -570,6 +570,49 @@ pub struct FirecrackerMetrics {
     pub uart: SerialDeviceMetrics,
     /// Metrics related to signals.
     pub signals: SignalMetrics,
+    /// Metrics related to virtio-vsockets.
+    pub vsock: VsockDeviceMetrics,
+}
+
+/// Vsock-related metrics.
+#[derive(Default, Serialize)]
+pub struct VsockDeviceMetrics {
+    /// Number of times when activate failed on a vsock device.
+    pub activate_fails: SharedMetric,
+    /// Number of times when interacting with the space config of a vsock device failed.
+    pub cfg_fails: SharedMetric,
+    /// Number of times when handling queue events on a vsock device failed.
+    pub queue_event_fails: SharedMetric,
+    /// Number of times when handling muxer events on a vsock device failed.
+    pub muxer_event_fails: SharedMetric,
+    /// Number of times when handling connection events on a vsock device failed.
+    pub conn_event_fails: SharedMetric,
+    /// Number of events associated with the receiving queue.
+    pub rx_queue_event_count: SharedMetric,
+    /// Number of events associated with the transmitting queue.
+    pub tx_queue_event_count: SharedMetric,
+    /// Number of bytes received.
+    pub rx_bytes_count: SharedMetric,
+    /// Number of transmitted bytes.
+    pub tx_bytes_count: SharedMetric,
+    /// Number of packets received.
+    pub rx_packets_count: SharedMetric,
+    /// Number of transmitted packets.
+    pub tx_packets_count: SharedMetric,
+    /// Number of added connections
```

Comment:
nit nit: can we be consistent here and use full stop for all the comments?

Reply:
Done